sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -454,6 +490,43 @@ case class BroadcastHashJoinExec(
     val (matched, checkCondition, _) = getJoinCondition(ctx, input)
     val numOutput = metricTerm(ctx, "numOutputRows")
 
+    if (isNullAwareAntiJoin) {
+      if (broadcastRelation.value.isInstanceOf[EmptyHashedRelation]) {
+        return s"""
+                  |// NAAJ Join EmptyHashedRelation accept all
+                  |$numOutput.add(1);
+                  |${consume(ctx, input)}
+            """.stripMargin
+      } else if (broadcastRelation.value.isInstanceOf[EmptyHashedRelationWithAllNullKeys]) {
+        return s"""
+                  |// NAAJ
+                  |// EmptyHashedRelationWithAllNullKeys
+                  |// reject all
```

Comment:
`// If the right side contains any all-null key, NAAJ simply returns Nil.`

Reply:
done